apps/meteor/server/routes/avatar/user.ts

```diff
@@ -0,0 +1,169 @@
+import type { ServerResponse } from 'http';
+
+import type { IUpload, IIncomingMessage } from '@rocket.chat/core-typings';
+import { Avatars, Users } from '@rocket.chat/models';
+import { serverFetch as fetch } from '@rocket.chat/server-fetch';
+import type { NextFunction } from 'connect';
+
+import { FileUpload } from '../../../app/file-upload/server';
+import { settings } from '../../../app/settings/server';
+import { renderSVGLetters, serveSvgAvatarInRequestedFormat, wasFallbackModified, setCacheAndDispositionHeaders } from './utils';
+
+const MAX_USER_SVG_AVATAR_SIZE = 1024;
+const MIN_USER_SVG_AVATAR_SIZE = 16;
+
+const serveSvgAvatar = (nameOrUsername: string, size: number, format: string, res: ServerResponse) => {
+	const svg = renderSVGLetters(nameOrUsername, size);
+	serveSvgAvatarInRequestedFormat(svg, format, res);
+};
+
+const getAvatarSizeFromRequest = (req: IIncomingMessage) => {
+	const requestSize = req.query.size && parseInt(req.query.size);
+	return Math.min(Math.max(requestSize, MIN_USER_SVG_AVATAR_SIZE), MAX_USER_SVG_AVATAR_SIZE);
+};
+
+const serveAvatarFile = (file: IUpload, req: IIncomingMessage, res: ServerResponse, next: NextFunction) => {
+	res.setHeader('Content-Security-Policy', "default-src 'none'");
+	const reqModifiedHeader = req.headers['if-modified-since'];
+
+	if (reqModifiedHeader && reqModifiedHeader === file.uploadedAt?.toUTCString()) {
+		res.setHeader('Last-Modified', reqModifiedHeader);
+		res.writeHead(304);
+		res.end();
+		return;
+	}
+
+	res.setHeader('Last-Modified', `${file.uploadedAt?.toUTCString()}`);
+	res.setHeader('Content-Type', file.type || '');
+	res.setHeader('Content-Length', file.size || 0);
+
+	return FileUpload.get(file, req, res, next);
+};
+
+const handleExternalProvider = async (externalProviderUrl: string, username: string, res: ServerResponse): Promise<boolean> => {
+	const response = await fetch(externalProviderUrl.replace('{username}', username));
+	response.headers.forEach((value, key) => res.setHeader(key, value));
+	response.body.pipe(res);
+	return true;
```

Comment:
The 2 places where this is called don't care about the return value, we can omit the return and make the func return void